### DISCUSSION TITLE
Can't get Fx layer to work

### DISCUSSION THREAD

**Discussion by babeyfaerie** [VolcanicArts/VRCOSC#15] [Q/A] [answered]
I've gotten all of the steps done, I just can't get the transition to work, no matter what I've tried. The heart rate things won't show up, so the toggle to toggle the heart rate doesn't work, so the numbers are always zero. I'm unsure what I did wrong

**Comment by VolcanicArts** [comment, marked as answer]
If you're not aware of it, try copying the animator layers and parameters using https://github.com/VRLabs/Avatars-3.0-Manager. Should make it easier to just copy everything from my example animator. Then all you have to do is setup your avatar state parameters exactly the same.

If you've got the HypeRate module enabled, also check the debug menu to see if the animator parameters are changing. If they're not, remember to reset your OSC config by going into your action menu under `Options` -> `OSC` -> `Reset Config`.

One thing to also mention is the `HeartrateEnabled` parameter is controlled by VRCOSC, and the `Heartrate` parameter is the one for doing the toggle override. It's not very clear in the guide so I will update it asap.

**Reply by babeyfaerie** [reply]
i added that tool to my project and it fixed my issue with that, however im not sure what avatar state parameters are, but i double checked my osc and everything is running as it should, input and output look correct, but the actual display on the avatar doesn't change. thank you for your help

**Reply by VolcanicArts** [reply]
Just to double check, you have placed the prefab on the root of your avatar, right?

Also your state parameters are what you add/remove on your avatar descriptor. They'll be next to where your menu is defined

**Reply by babeyfaerie** [reply]
when you say root, do you mean in the armature? i have it in the armature on her chest, and parameters as in the parameters list with the bools and the floats? i added the parameters from the parameter menu and the actual toggle from the other menu

**Reply by VolcanicArts** [reply]
It needs to be placed as a child of the object that has your avatar descriptor (effectively alongside your armature) otherwise the animations will not work. Then you can use a parent constraint to attach it to your chest.

**Reply by babeyfaerie** [reply]
im not sure how to do that, im still learning unity, im very sorry
do you mean place it next to the armature in the hierarchy?

**Reply by VolcanicArts** [reply]
Okay, so. Move the heartrate prefab onto your avatar by dragging it onto it.

In your inspector, on the heartrate prefab, click `add component` and type `parent constraint`

Then add a source and set the source to your chest.

Ensure `isActive` is off. Move the prefab to where you want and press `activate`

And yes, next to the armature

**Reply by babeyfaerie** [reply]
its working now thanks so much!

**Reply by VolcanicArts** [reply]
No problem. Enjoy!

I'm going to put together an FAQ post for people that have the same problems as the documentation for the prefabs is lacking in detail about where to place the prefab.
